test/storage.test.js

```diff
@@ -17,8 +19,22 @@ describe('Storage tests for feeds', () => {
   const feed2 = new Feed('James Smith 2', 'http://seneca.co/jsmith/2');
   const feed3 = new Feed('James Smith 2', 'http://seneca.co/jsmith/3', 'etag');
   const feed4 = new Feed('James Smith 2', 'http://seneca.co/jsmith/4', 'etag', 'last-modified');
-
-  beforeAll(() => Promise.all([addFeed(feed1), addFeed(feed2), addFeed(feed3), addFeed(feed4)]));
+  const feed5 = new Feed(
+    'James Smith Invalid Feed',
+    'http://seneca.co/jsmith/4',
+    'etag',
+    'last-modified'
+  );
+
+  beforeAll(() =>
+    Promise.all([
+      addFeed(feed1),
+      addFeed(feed2),
+      addFeed(feed3),
+      addFeed(feed4),
+      setInvalidFeed(feed5.id, 'This just fails'),
```

Comment:
You haven't added `feed5` to Redis, so this can't work.  You can only invalidate a feed that exists in the db.

Reply:
We have mock-redis so I'm guessing I can save this in mock-redis and it should be fine?

Reply:
Correct, you can do whatever you want in the test code, it's all being done in-memory with the mock redis, so nothing will get broken.

Reply:
Probably `setInvalidFeed` should check that the feed exists, and throw if it doesn't.  We should deal with that problem higher up in the call stack.

I'm unclear how a feed could not be in our feed set in Redis.  Can you trace this back, and make sure that we're adding the feed before we start to rely on it?

Reply:
In the `processor.js` file
the first if statement with 
```
if (invalid) {
//code
}
```
should always be skipped at the start, because there shouldn't be any invalid feeds in the db to retrieve with the `GET` command. Otherwise later on in the code you have the line `await feed.save()` which actually will save the feed to Redis

I think the way I implemented your solution though might be wrong. We don't actually need a feed to be saved to the database to get a key,

In `storage.js` this is the following code to set an invalid feed
```
  setInvalidFeed: (id, reason) => {
    const key = createInvalidFeedKey(id);
    redis.set(key, reason);
  }
```
as long as an ID and a reason is provided to the function, Redis will create a key value pair using the `SET` command.

Reply:
Right, so we'll always process a feed once, which will `feed.save()` it into Redis before we can possibly try and invalidate it.  Good.

I want to avoid orphaned feed keys, where we create some invalid key for a feed that doesn't exist.  It's not a problem per se, but it just wastes space in the db with something we'll never find again.